### PR TITLE
[DT-8] feat: improve error message on reports limit reached

### PR DESCRIPTION
**Pull request by JoanVicens**
When the report limit for the day has reached the message shown was for a generic error.

### CODE REVIEW

File: src/main/bug-report/BugReportResult.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+type Ok = { state: 'OK' };
+type Bad = { state: 'ERROR' | 'TO_MANY_REPORTS' };
```
Comment (sg-gs) [inline src/main/bug-report/BugReportResult.ts:2]: `export type BadReportResult = ..`

File: src/main/bug-report/service.ts [hunk 2]
```diff
@@ -36,7 +37,16 @@ export async function sendReport({
...
     headers: { Authorization: `Bearer ${getToken()}` },
   });
-  if (!res.ok) throw new Error();
+
+  if (res.ok) return { state: 'OK' };
+
+  log.error(`Report status: ${res.status}`);
+
+  if (res.status === 429) {
+    return { state: 'TO_MANY_REPORTS' };
```
Comment (sg-gs) [inline src/main/bug-report/service.ts:46]: `TOO_MANY_REPORTS`

File: src/renderer/pages/ProcessIssues/ReportModal.tsx [hunk 3]
```diff
@@ -5,6 +5,20 @@ import { longMessages, shortMessages } from '../../messages/process-error';
 import Spinner from '../../assets/spinner.svg';
 import { ProcessIssue } from '../../../workers/types';
 
+const posibleErrorStates = ['ERROR', 'TO_MANY_REPORTS'] as const;
```
Comment (sg-gs) [inline src/renderer/pages/ProcessIssues/ReportModal.tsx:8]: `TOO_MANY_REPORTS`

File: src/renderer/pages/ProcessIssues/ReportModal.tsx [hunk 4]
```diff
@@ -5,6 +5,20 @@ import { longMessages, shortMessages } from '../../messages/process-error';
...
+type ReportRequestState = 'READY' | 'SENDING' | ErrorReportRequestState;
+
+const stateIsError = (maybe: unknown): maybe is ErrorReportRequestState =>
+  typeof maybe === 'string' &&
+  posibleErrorStates.includes(maybe as ErrorReportRequestState);
+
+const errorMessages: Record<ErrorReportRequestState, string> = {
+  ERROR:
+    'We could not send your request, make sure you are connected to the internet',
+  TO_MANY_REPORTS: 'You cannot create more request for today',
```
Comment (sg-gs) [inline src/renderer/pages/ProcessIssues/ReportModal.tsx:19]: `TOO_MANY_REPORTS`

Comment (sg-gs) [reply src/renderer/pages/ProcessIssues/ReportModal.tsx:19]: And a better message could be "Reached reporting errors day limit"

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/main/bug-report/BugReportResult.ts (hunk 1 above)
Comment (JoanVicens) [reply src/main/bug-report/BugReportResult.ts]: done

File: src/renderer/pages/ProcessIssues/ReportModal.tsx (hunk 3 above)
Comment (JoanVicens) [reply src/renderer/pages/ProcessIssues/ReportModal.tsx]: done

File: src/renderer/pages/ProcessIssues/ReportModal.tsx (hunk 4 above)
Comment (JoanVicens) [reply src/renderer/pages/ProcessIssues/ReportModal.tsx]: done

**Pull request merged**